### ISSUE TITLE
Feature Request: Considering handling a SkipEnqueue MiddlewareError

### ISSUE THREAD

**Issue by dperetti** [Bogdanp/dramatiq#276]
I've just implemented a "DedupMiddleware" which can be used prevent a message from being enqueued if a similar message already exists in the queue.
Unfortunately it requires to override `RedisBroker.enqueue()` in order to gracefully catch the `SkipEnqueue` middleware error I throw when the message needs.

Would you consider accepting a PR that would:
1. Define a `SkipEnqueue(MiddlewareError)`.
2. `try..catch` this error around self.emit_before [here](https://github.com/Bogdanp/dramatiq/blob/abbbc20464ca83a380e6063711fd2e6aaf3a9b10/dramatiq/brokers/redis.py#L175), so a middleware could be given the possibility to interrupt an enqueuing in that hook?

Thanks.

**Comment by Bogdanp** [comment]
Yes, I would accept this as long as you update all the brokers and document the behaviour.
